metals/src/main/scala/scala/meta/internal/metals/testProvider/frameworks/MunitTestFinder.scala

```diff
@@ -0,0 +1,150 @@
+package scala.meta.internal.metals.testProvider.frameworks
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+import scala.meta.Defn
+import scala.meta.Lit
+import scala.meta.Pkg
+import scala.meta.Term
+import scala.meta.Tree
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.testProvider.FullyQualifiedName
+import scala.meta.internal.metals.testProvider.TestCaseEntry
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+
+class MunitTestFinder(trees: Trees) {
+
+  def findTests(
+      path: AbsolutePath,
+      suiteName: FullyQualifiedName
+  ): Vector[TestCaseEntry] = {
+    val uri = path.toURI
+    val testcases = new mutable.ArrayBuffer[TestCaseEntry]()
+
+    /**
+     * Class definition is valid when package + class name is equal to one we are looking for
+     */
+    def isValid(cls: Defn.Class, currentPackage: String): Boolean =
+      s"$currentPackage.${cls.name.value}" == suiteName.value
+
+    def loop(tree: Tree, currentPackage: Vector[String]): Unit = {
+      val pkgString = currentPackage.mkString(".")
+      tree match {
+        case cls: Defn.Class if isValid(cls, pkgString) =>
+          /**
+           * In munit, it's very popular to define helper method for tests
+           * which prevents from code duplication. These method often looks like:
+           * def check(name: String, ...) = {
+           *   test(name) {
+           *     <test logic>
+           *   }
+           * }
+           * Finding these potential test methods will allow to show them to the user.
+           */
+          val potentialTests = cls.templ.children.collect {
+            // def check(...) = { test("") {} }
+            case Defn.Def(
+                  _,
+                  name,
+                  _,
+                  _,
+                  _,
+                  Term.Block(
+                    List(Term.Apply(Term.Apply(Term.Name("test"), _), _))
+                  )
+                ) =>
+              name.value
+            // def check(...) = test("") {}
+            case Defn.Def(
+                  _,
+                  name,
+                  _,
+                  _,
+                  _,
+                  Term.Apply(Term.Apply(Term.Name("test"), _), _)
+                ) =>
+              name.value
+          }.toSet
+
+          // let's collect all tests candidates
+          cls.templ.children.collect {
+            // test("test1") {}
```

Comment:
This seems a bit limited. Wouldn't it be better to additionally use semanticdb here?
We could just look for 
```
Term.Apply(
  test @ Term.Name("test"),
  _
)
```
And compare the range on the name with semanticdb to see if it corresponds to the munit method. This way we would handle all possible configurations.

We could also reuse the function to search the children of `Defn.Def` for invocations of test and this might less error prone.